hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1947,6 +1947,9 @@ public boolean shouldPerformMajorCompaction() throws IOException {
     if (!this.areWritesEnabled()) {
       return Optional.empty();
     }
+    if (this.getCoprocessorHost() != null) {
```

Comment:
Is this the right place to add this CP call?

Reply:
Move this check to `RSRpcServices`. Internal cause compaction not need check permission

Reply:
Will this introduce a behavior change? We used to check it in preCompact, so for non rpc compactions we will also do the check?

Reply:
so, the former change, check permission in `Hstore.requestCompaction` will do check for non rpc compactions

Reply:
Do you think we need to check for non rpc compactions? I do not mean we must keep the old behavior, but we need a reason to change it.

Reply:
The original permission check in `preCompact` will check both rpc and non rpc compactions, If we move the check to  `Hstore.requestCompaction` we can check both rpc and non rpc compactions also.